test.js

```diff
@@ -69,4 +75,11 @@ test('unist-util-map', function (t) {
   function addValue() {
     return {value: 'test'}
   }
+
+  /**
+   * @type {import('./index.js').MapFunction<AnyNode>}
```

Comment:
```suggestion
   * @type {import('./index.js').MapFunction<Root>}
```

The type parameter to `map` can be a single root node. `MapFunction` infers which children are in that tree.